beacon-chain/core/blocks/block_operations.go

```diff
@@ -659,6 +673,18 @@ func ProcessDeposits(
 ) (*pb.BeaconState, error) {
 	var err error
 	deposits := body.Deposits
+	// Verify that outstanding deposits are processed up to the maximum number of deposits.
+	maxDeposits := beaconState.Eth1Data.DepositCount - beaconState.Eth1DepositIndex
```

Comment:
We don't need it here, it's checked as part of `process_operation`:
https://github.com/prysmaticlabs/prysm/blob/spec-v0.6/beacon-chain/core/state/transition.go#L262

Reply:
Why was it moved?

Reply:
It's not removed, it's moved to process_operation

Reply:
so we don't have to process proposer slashings, attester slashings and attestations (1/2 of the process_block cycle) before realizing the deposit amount is incorrect

I can address this in my block operations PR if you need to merge this urgently